static/js/disaster_dashboard/data_fetcher.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Functions that return promises for data needed for disaster dashbaord
+ */
+
+import axios from "axios";
+
+import { GeoJsonData } from "../chart/types";
+import { DISASTER_EVENT_INTENSITIES, DISASTER_EVENT_TYPES } from "./constants";
+import { DisasterApiEventPoint, DisasterEventPoint, DisasterType } from "./types";
+
+const MAX_DATES = 100;
+
+/**
+ * Get promise for geojson data
+ * @param selectedPlace the enclosing place to get geojson data for
+ * @param placeType the place type to get geojson data for
+ */
+export function fetchGeoJsonData(
+  selectedPlace: string,
+  placeType: string
+): Promise<GeoJsonData> {
+  return axios
+    .get<GeoJsonData>("/api/choropleth/geojson", {
+      params: {
+        placeDcid: selectedPlace,
+        placeType: placeType,
+      },
+    })
+    .then((resp) => resp.data as GeoJsonData);
+}
+
+/**
```

Comment:
If you have time, maybe add a simple test for each of the function here, which can be used to guard any future data change. Also could see what the data looks like for each promise.

Reply:
if adding a client side test, I would probably be mocking the data that comes back from the API, so I'm not sure that would guard data changes? Or would the test be to make sure the merging of multiple data responses works?

Reply:
Mocking data from API works. That can help any change in the Flask API or the data merge logic here works. If that's too much given the time constraint, maybe setup the test and do it for one of the functions here as an example.

Reply:
done

Reply:
thanks!